openid_connect.js

```diff
@@ -7,6 +7,36 @@ var newSession = false; // Used by oidcAuth() and validateIdToken()
 
 export default {auth, codeExchange, validateIdToken, logout};
 
+
+function authZUriHandler(r) {
+    // Choose a nonce for this flow for the client, and hash it for the IdP
+    var authZUri = null;
+    var noncePlain = r.variables.request_id;
+    var c = require('crypto');
+    var h = c.createHmac('sha256', r.variables.oidc_hmac_key).update(noncePlain);
+    var nonceHash = h.digest('base64url');
+
+    // Redirect the client to the IdP login page with the cookies we need for state
+    r.headersOut['Set-Cookie'] = [
+        "auth_redir=" + r.variables.request_uri + "; " + r.variables.oidc_cookie_flags,
+        "auth_nonce=" + noncePlain + "; " + r.variables.oidc_cookie_flags ];
+
+    if ( r.variables.oidc_pkce_enable == 1 ) {
```

Comment:
`s/==/===` ?

Reply:
NGINX handles variables always as strings. Therfore `===` will not work here. Possible solution would be something like `String(r.variables.oidc_pkce_enable) === "1"`. But looks a little bit to hacky to me.